pkg/proxy/nftables/proxier.go

```diff
@@ -328,6 +328,7 @@ type nftablesBaseChain struct {
 var nftablesBaseChains = []nftablesBaseChain{
 	// We want our filtering rules to operate on pre-DNAT dest IPs, so our filter
 	// chains have to run before DNAT.
+	{"filter-prerouting", knftables.FilterType, knftables.PreroutingHook, knftables.DNATPriority + "-1"},
 	{"filter-input", knftables.FilterType, knftables.InputHook, knftables.DNATPriority + "-1"},
 	{"filter-forward", knftables.FilterType, knftables.ForwardHook, knftables.DNATPriority + "-1"},
 	{"filter-output", knftables.FilterType, knftables.OutputHook, knftables.DNATPriority + "-1"},
```

Comment:
I set those priorities thinking that priority had precedence over hook; if it doesn't then I think we can just use `knftables.FilterPriority` for all the filter chains... I need to dig into this more.

(Also, not sure we need filter-input/filter-output any more? We could probably just move `kubeEndpointsCheckChain` over to filter-prerouting too?)

Reply:
OK, we do still need to set the priorities here, because we need `filter-prerouting` to run before `nat-prerouting` and `filter-output` to run before `nat-output`. While we're here though, let's change them from "DNATPriority - 1" to "DNATPriority - 10", to theoretically allow third parties to insert something else between them...


> (Also, not sure we need filter-input/filter-output any more? We could probably just move `kubeEndpointsCheckChain` over to filter-prerouting too?)

ISTM that `kubeEndpointsCheckChain` would have exactly the same effect running from `prerouting`+`output` as it would running from `input`+`forward`+`output`. In iptables we had to do this from `INPUT`+`FORWARD`+`OUTPUT` because there's no `PREROUTING` chain in `filter`, but in nftables we don't have that restriction. So yeah, I think you can drop `filter-input` and `filter-forward`.

(Well, OK, I guess we still need `filter-forward` for `kubeForwardChain`... but we should get rid of that... but that's another PR.)

Reply:
done, removed `filter-input` and kept `filter-forward` for now.
priorities changed.

Reply:
Just found we can do the same for `kubeEndpointsCheckChain` because its uses `reject`, which can [only be used in `input`, `forward` and `output`](https://www.netfilter.org/projects/nftables/manpage.html): 

> A reject statement is used to send back an error packet in response to the matched packet otherwise it is equivalent to drop so it is a terminating statement, ending rule traversal. **This statement is only valid in base chains using the input, forward or output hooks, and user-defined chains which are only called from those chains**.

The error got when running:
```
E0108 11:35:20.657740       1 proxier.go:1550] "nftables sync failed" err=<
        /dev/stdin:59:1-128: Error: Could not process rule: Operation not supported
        add element ip kube-proxy no-endpoint-services { 10.96.0.10 . udp . 53 comment "kube-system/kube-dns:dns" : goto reject-chain }
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        /dev/stdin:61:1-134: Error: Could not process rule: Operation not supported
        add element ip kube-proxy no-endpoint-services { 10.96.0.10 . tcp . 9153 comment "kube-system/kube-dns:metrics" : goto reject-chain }
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        /dev/stdin:60:1-132: Error: Could not process rule: Operation not supported
        add element ip kube-proxy no-endpoint-services { 10.96.0.10 . tcp . 53 comment "kube-system/kube-dns:dns-tcp" : goto reject-chain }
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Reply:
oh, interesting. They must have relaxed this at some point because it works on Fedora 39 (kernel 6.6 / nft 1.0.7).

Reply:
You are right. It's supported since 5.9: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=f53b9b0bdc59c0823679f2e3214e0d538f5951b9. Sent a patch to nftables to correct the misleading doc.